### PR TITLE
Fix HERP filter time zone issue.

### PR DESCRIPTION
**Pull request by wdschei**

### CODE REVIEW

**Comment by wdschei** [comment]
Even better this is running on master which is the host that caused this issue to show up in the first place.

File: repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala [hunk 1]
```diff
@@ -210,7 +210,15 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
 
 class CadfTimestamp extends Helper[Long] {
   override def apply(context: Long, options: Options): CharSequence = {
-    new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX").format(new Date(context))
+    // From http://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html
+    // "For formatting, if the offset value from GMT is 0, 'Z' is produced."
+    // This only manipulates the Time Zones that produce a 'Z'.
+    val formattedString = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX").format(new Date(context))
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: Is this format intentionally different than the original?

Comment (wdschei) [inline repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: Yes. This adds the fractional second.

**Comment by wdschei** [comment]
The Spock tests that failed need the same update to account for the zero offset from GMT producing only a 'Z'.

**Comment by wdschei** [comment]
Retest this please.

**Comment by wdschei** [comment]
Retest this please.

**Comment by wdschei** [comment]
Retest this please.

**Comment by wdschei** [comment]
Retest this please.

**Comment by wdschei** [comment]
Retest this please.

**Comment by wdschei** [comment]
Retest this please.

**Comment by wdschei** [comment]
Retest this please.

**Pull request merged**